xonsh/dirstack.py

```diff
@@ -15,14 +15,24 @@ def _get_cwd():
         return None
 
 
+def _splitpath(path, sofar=[]):
```

Comment:
The default value shouldn't be mutable, due to Python's side effect issues in this case.